release.sh

```diff
@@ -0,0 +1,464 @@
+
+  confirm "Create tag $version?"
+  run git tag $version
+
+  confirm "Push tag and commit to GitHub?"
+  run git push
+  run git push --tags
+
+  revert_repo
+
+  echo "🍺  Tag $version created!"
+}
+
+function release {
+
+  confirm "Create a new release for $version?"
+
+  message "Stashing current work..."
+
+  stash_ref=$(git stash create)
+  run_message git stash create
+
+  if [[ $stash_ref ]]; then
+    run git reset --hard
+    message "Stashed current work to: $stash_ref"
+  else
+    message "Nothing to stash"
+  fi
+
+  branch_name="$(git symbolic-ref HEAD 2>/dev/null)"
+  branch_name=${branch_name##refs/heads/}
+  message "Current branch: $branch_name"
+
+  if [[ $branch_name != 'master' ]]; then
+    message "Switching to master..."
+    run git checkout master
+  fi
+
+  message "Pulling latest version from GitHub"
+  run git pull --rebase
+
+  if [[ ! -e resty ]]; then
+    message "Downloading resty to connect to GitHub API"
+    run curl -L http://github.com/micha/resty/raw/master/resty > resty
+  fi
+  if [[ ! -e jsawk ]]; then
+    message "Downloading jsawk to parse info from GitHub API"
+    run curl -L http://github.com/micha/jsawk/raw/master/jsawk > jsawk
+  fi
+
+  prompt "GitHub access-token (required only for 2fac):"
+  github_access_token=$user_response
+
+  if [[ github_access_token ]]; then
+    run . resty -W 'https://api.github.com' -H "Authorization: token $github_access_token"
+  else
+    prompt "GitHub username:"
+    github_username=$user_response
+
+    prompt_password "GitHub password:"
+    github_password=$user_response
+
+    run . resty -W 'https://api.github.com' -u $github_username:$github_password
+  fi
+
+  response=$(run POST /repos/Automattic/facebook-instant-articles-wp/releases "
+  {
+    \"tag_name\": \"$version\",
+    \"target_commitish\": \"master\",
+    \"name\": \"$version\",
+    \"body\": \"Version $version\",
+    \"draft\": false,
+    \"prerelease\": false
+  }
+  ");
+
+  if [[ $response ]]; then
+    message "Release $version created!"
+  else
+    error_message "Couldn't create release"
+  fi
+
+  upload_url=$( echo $response | . jsawk -n 'out(this.upload_url)' |  sed -e "s/{[^}]*}//g" )
+  release_id=$( echo $response | . jsawk -n 'out(this.id)'  )
+
+  message "Upload URL: $upload_url"
+
+  message "Creating binary file"
+  run composer install
+  run zip -qr facebook-instant-articles-wp.zip .
+
+  message "Uploading binary for release..."
+
+  if [[ github_access_token ]]; then
+    response=$(run curl -H "Authorization: token $github_access_token" -H "Content-Type: application/zip" --data-binary @facebook-instant-articles-wp.zip $upload_url\?name=facebook-instant-articles-wp-$version.zip )
+  else
+    response=$(run curl -u $github_username:$github_password -H "Content-Type: application/zip" --data-binary @facebook-instant-articles-wp.zip $upload_url\?name=facebook-instant-articles-wp-$version.zip )
+  fi
+
+  run rm facebook-instant-articles-wp.zip
```

Comment:
For simplicity I wouldn't keep the zip file even if the upload wasn't successful. Reason being I want it to always create the file again in order to avoid dealing with the invalidation of the file (due to changes on the repo).